x-pack/plugins/security_solution/server/usage/detections_helpers.ts

```diff
@@ -0,0 +1,156 @@
+      detection_rules_elastic_disabled: usage.detection_rules_elastic_disabled + 1,
+    };
+  } else if (isEnabled && !isElastic) {
+    return {
+      ...usage,
+      detection_rules_custom_enabled: usage.detection_rules_custom_enabled + 1,
+    };
+  } else if (!isEnabled && !isElastic) {
+    return {
+      ...usage,
+      detection_rules_custom_disabled: usage.detection_rules_custom_disabled + 1,
+    };
+  } else {
+    return usage;
+  }
+};
+
+const updateMlJobsUsage = (jobMetric: DetectionsMetric, usage: MlJobsUsage): MlJobsUsage => {
+  const { isEnabled, isElastic } = jobMetric;
+  if (isEnabled && isElastic) {
+    return {
+      ...usage,
+      ml_jobs_elastic_enabled: usage.ml_jobs_elastic_enabled + 1,
+    };
+  } else if (!isEnabled && isElastic) {
+    return {
+      ...usage,
+      ml_jobs_elastic_disabled: usage.ml_jobs_elastic_disabled + 1,
+    };
+  } else if (isEnabled && !isElastic) {
+    return {
+      ...usage,
+      ml_jobs_custom_enabled: usage.ml_jobs_custom_enabled + 1,
+    };
+  } else if (!isEnabled && !isElastic) {
+    return {
+      ...usage,
+      ml_jobs_custom_disabled: usage.ml_jobs_custom_disabled + 1,
+    };
+  } else {
+    return usage;
+  }
+};
+
+export const getRulesUsage = async (
+  index: string,
+  callCluster: LegacyAPICaller
+): Promise<DetectionRulesUsage> => {
+  let rulesUsage: DetectionRulesUsage = initialRulesUsage;
+  const ruleSearchOptions: SearchParams = {
+    body: { query: { bool: { filter: { term: { 'alert.alertTypeId': SIGNALS_ID } } } } },
+    filterPath: ['hits.hits._source.alert.enabled', 'hits.hits._source.alert.tags'],
+    ignoreUnavailable: true,
+    index,
+    size: 10000, // elasticsearch index.max_result_window default value
+  };
+
+  try {
+    const ruleResults = await callCluster<{ alert: { enabled: boolean; tags: string[] } }>(
+      'search',
+      ruleSearchOptions
+    );
+
+    if (ruleResults.hits?.hits?.length > 0) {
+      rulesUsage = ruleResults.hits.hits.reduce((usage, hit) => {
+        const isElastic = isElasticRule(hit._source.alert.tags);
+        const isEnabled = hit._source.alert.enabled;
+
+        return updateRulesUsage({ isElastic, isEnabled }, usage);
+      }, initialRulesUsage);
+    }
+  } catch (e) {
+    // ignore failure, usage will be zeroed
+  }
+
+  return rulesUsage;
+};
+
+export const getMlJobsUsage = async (ml: MlPluginSetup | undefined): Promise<MlJobsUsage> => {
+  let jobsUsage: MlJobsUsage = initialMlJobsUsage;
+
+  if (ml) {
+    try {
+      const mlCaller = ml.mlClient.callAsInternalUser;
+      const modules = await new DataRecognizer(
+        mlCaller,
+        ({} as unknown) as SavedObjectsClient
+      ).listModules();
+      const moduleJobs = modules.flatMap((module) => module.jobs);
+      const jobs = await jobServiceProvider(mlCaller).jobsSummary(['siem']);
+
+      jobsUsage = jobs.reduce((usage, job) => {
+        const isElastic = moduleJobs.some((moduleJob) => moduleJob.id === job.id);
```

Comment:
does SIEM apply a prefix to the module when initially setting it up?
If so, this won't match and you'll have to do a partial match on the job id.

Reply:
Confirmed that this is not currently an issue, but good to know moving forward.